pkg/engine/internal/executor/functions.go

```diff
@@ -204,6 +224,47 @@ func (f *genericFunction[E, T]) Evaluate(lhs ColumnVector, rhs ColumnVector) (Co
 	return &Array{array: builder.NewArray()}, nil
 }
 
+// genericFloat64Function is a struct that implements the [BinaryFunction] interface methods
+// and can be used for any array type with numeric elements.
+type genericFloat64Function[E arrayType[T], T comparable] struct {
+	eval func(a, b T) (float64, error)
+}
```

Comment:
Could this be expressed as a generic function


```suggestion
type genericFunction[E arrayType[T], T comparable, R any] struct {
	eval func(a, b T) (R, error)
}
```

Reply:
I have not found a clean way to do that in a generic way. There are some pretty specific type coercions based on the result type, plus `pow` and `mod` are implemented totally differently.